prolif/io/protein_helper.py

```diff
@@ -0,0 +1,594 @@
+        The ProLIF Molecule instance representing the protein.
+
+    Notes
+    -----
+    This class is designed to work with ProLIF Molecule instances or PDB files.
+    It reads the input topology, standardizes the residue names, and fixes the bond
+    orders based on the provided templates or standard amino acid template.
+    However, the explicit hydrogen atoms will be removed from the molecule.
+
+    Example
+    -------
+    >>> from prolif.io import ProteinHelper
+    >>> protein_helper = ProteinHelper("path/to/protein.pdb")
+    >>> protein_helper.standardize_protein()
+    >>> # Now, protein_helper.protein_mol contains the standardized protein molecule.
+    >>> # You can access the residues using protein_helper.protein_mol.residues.
+
+    """
+
+    def __init__(self, input_topology: Molecule | str | Path):
+        # read as prolif molecule
+        if isinstance(input_topology, Molecule):
+            self.protein_mol = input_topology
+
+        elif isinstance(input_topology, str | Path) and str(input_topology).endswith(
+            ".pdb"
+        ):
+            input_protein_top = Chem.MolFromPDBFile(str(input_topology))
+            self.protein_mol = Molecule.from_rdkit(input_protein_top)
+
+        else:
+            raise TypeError(
+                "input_topology must be a string (path to a PDB file) or "
+                "a prolif Molecule instance."
+            )
+
+    def standardize_protein(self, templates: list[dict] | dict | None = None) -> None:
+        """Standardize the protein molecule.
+
+        This function will standardize the residue names, fix the bond orders,
+        and check the residue names against the templates.
+
+        Parameters
+        ----------
+        templates : list[dict] or dict or None, optional
+            The templates to use for standardizing the protein molecule.
+            If `None`, the standard amino acid template is used.
+            If a dict is provided, it should contain the templates for residues.
+            If a list is provided, it should contain dictionaries for each template.
+            Default is `None`.
+
+        Example
+        -------
+        >>> from prolif.io import ProteinHelper
+        >>> protein_helper = ProteinHelper("path/to/protein.pdb")
+        >>> protein_helper.standardize_protein(
+                templates=[{"ALA": {"SMILES": "CC(C(=O)O)N"}}]
+            )
+        """
+
+        # read the templates
+        if templates is None:
+            templates = [STANDARD_AA]
+        elif isinstance(templates, dict):
+            # if templates is a dict, convert it to a list of dicts
+            templates = [templates, STANDARD_AA]
+        elif isinstance(templates, list):
+            # if templates is a list, check if it contains dicts
+            if not all(isinstance(t, dict) for t in templates):
+                raise TypeError("Templates must be a dict, a list of dicts or None.")
+            templates.extend([STANDARD_AA])
```

Comment:
Best to avoid directly modifying the list provided by the user:
```suggestion
            templates = [*templates, STANDARD_AA]
```